addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -615,9 +614,37 @@ var ImageWidget = MediaWidget.extend({
     _onSearchInput: function (ev) {
         var $input = $(ev.currentTarget);
         var $button = $input.next('.input-group-append').children();
-        var emptyValue = ($input.val() === '');
+        var inputValue = $input.val();
+        var emptyValue = (inputValue === '');
+        var $warning = this.$('.warning-url');
+        var $success = this.$('.success-url');
+        var $error = this.$('.error-url');
+        var $btn = this.$('.o_upload_media_url_button');
+        var regex = /^(?:https?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/;
+        var testURL = regex.test(inputValue);
+        var displayNone = !testURL;
+        var emptyURL = inputValue.length !== 0;
+        var formats = ['.gif', '.jpe', '.jpg', '.png'];
```

Comment:
Do not declare the `formats` variable as it is only used once and only in some context. Do directly
```javascript
_.each(['.gif', '.jpe', '.jpg', '.png'], ...)
```
it is clearer, more efficient and it is less code.